src/components/hover-cards.vue

```diff
@@ -84,14 +84,25 @@ const component = shallowRef(null);
 const placement = ref(undefined);
 
 const popoverRef = ref(null);
-const positionPopover = () => {
-  const rightOfAnchor = hoverCard.anchor.getBoundingClientRect().right;
-  const componentElement = popoverRef.value.$el.children[0];
-  const componentWidth = componentElement.getBoundingClientRect().width;
-  // Subtracting 50 in order to provide some buffer.
-  const overflowsRight = rightOfAnchor + componentWidth >
-    document.documentElement.clientWidth - 50;
-  placement.value = overflowsRight ? 'left' : 'right';
+// The minimum distance in px between the edge of the hover card and the edge of
+// the viewport. This provides space for the popover arrow, as well some extra
+// buffer.
+const buffer = 20;
+const computePlacement = () => {
+  const anchorRect = hoverCard.anchor.getBoundingClientRect();
+  const popoverRect = popoverRef.value.$el.getBoundingClientRect();
+
+  // Try 'right' and 'left' first unless the anchor is so close to the bottom
+  // that the popover won't fit.
+  if (window.scrollY + anchorRect.bottom + (popoverRect.height / 2) <
```

Comment:
`/ 2` because the popover is vertically centered relative to the anchor element.

Reply:
I added a code comment explaining this.